### DISCUSSION TITLE
CentOS installing error.

### DISCUSSION THREAD

**Discussion by LITTLEKKKK** [libvips/libvips#2114] [Q&A]
when I do "sudo make", how to solve this problem?

make[3]: warning:  Clock skew detected.  Your build may be incomplete.
make[3]: Leaving directory `/public/software/libvips/vips-8.10.5/test'
make[2]: Leaving directory `/public/software/libvips/vips-8.10.5/test'
Making all in fuzz
make[2]: Entering directory `/public/software/libvips/vips-8.10.5/fuzz'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/public/software/libvips/vips-8.10.5/fuzz'
make[2]: Entering directory `/public/software/libvips/vips-8.10.5'
make[2]: Leaving directory `/public/software/libvips/vips-8.10.5'
make[1]: warning:  Clock skew detected.  Your build may be incomplete.
make[1]: Leaving directory `/public/software/libvips/vips-8.10.5'
make: warning:  Clock skew detected.  Your build may be incomplete.

**Comment by LITTLEKKKK** [comment]
make[3]: warning: Clock skew detected. Your build may be incomplete.

This problem I solved.

"make[2]: Leaving directory xxxx"
 how to solve this problem?

**Reply by jcupitt** [reply]
Hi, I'd guess your system clock is wrong. Set the time and try again from the beginning.

**Reply by LITTLEKKKK** [reply]
I have solved this problem. But "leaving directory" error still remind.

**Comment by LITTLEKKKK** [comment]
"./configure", this is the list below.


checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for gobject-introspection... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for needs -lstdc++... yes
checking for native Win32... no
checking for binary open needed... no
checking for Mac OS X... no
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for gawk... (cached) gawk
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for gcc option to accept ISO C99... -std=gnu99
checking for gcc -std=gnu99 option to accept ISO Standard C... (cached) -std=gnu99
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether ln -s works... yes
checking if malloc debugging is wanted... no
checking locale.h usability... yes
checking locale.h presence... yes
checking for locale.h... yes
checking for LC_MESSAGES... yes
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking libintl.h usability... yes
checking libintl.h presence... yes
checking for libintl.h... yes
checking for ngettext in libc... yes
checking for dgettext in libc... yes
checking for bind_textdomain_codeset... yes
checking for msgfmt... /usr/bin/msgfmt
checking for dcgettext... yes
checking if msgfmt accepts -c... yes
checking for gmsgfmt... /usr/bin/msgfmt
checking for xgettext... /usr/bin/xgettext
checking for catalogs to be installed...  en_GB de
checking for gcc version... 4.8.5
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for ANSI C header files... (cached) yes
checking errno.h usability... yes
checking errno.h presence... yes
checking for errno.h... yes
checking math.h usability... yes
checking math.h presence... yes
checking for math.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/file.h usability... yes
checking sys/file.h presence... yes
checking for sys/file.h... yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sys/mman.h usability... yes
checking sys/mman.h presence... yes
checking for sys/mman.h... yes
checking for sys/types.h... (cached) yes
checking for sys/stat.h... (cached) yes
checking for unistd.h... (cached) yes
checking io.h usability... no
checking io.h presence... no
checking for io.h... no
checking direct.h usability... no
checking direct.h presence... no
checking for direct.h... no
checking windows.h usability... no
checking windows.h presence... no
checking for windows.h... no
checking for dllwrap... no
checking for dlltool... false
checking for objdump... (cached) objdump
checking for ranlib... (cached) ranlib
checking for strip... (cached) strip
checking for ar... (cached) ar
checking for as... as
checking for ld... /usr/bin/ld -m elf_x86_64
checking for C/C++ restrict keyword... __restrict
checking for __attribute__((vector_size))... yes
checking for an ANSI C-conforming const... yes
checking for mode_t... yes
checking for off_t... yes
checking for size_t... yes
checking for gcc with working vector support... no
checking for working memcmp... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/param.h... (cached) yes
checking for getpagesize... yes
checking for working mmap... yes
checking for vprintf... yes
checking for _doprnt... no
checking for getcwd... yes
checking for gettimeofday... yes
checking for getwd... yes
checking for memset... yes
checking for munmap... yes
checking for putenv... yes
checking for realpath... yes
checking for strcasecmp... yes
checking for strchr... yes
checking for strcspn... yes
checking for strdup... yes
checking for strerror... yes
checking for strrchr... yes
checking for strspn... yes
checking for vsnprintf... yes
checking for realpath... (cached) yes
checking for mkstemp... yes
checking for mktemp... yes
checking for random... yes
checking for rand... yes
checking for sysconf... yes
checking for atexit... yes
checking for cbrt in -lm... yes
checking for hypot in -lm... yes
checking for atan2 in -lm... yes
checking for REQUIRED... yes
checking for MONOTONIC_TIME... yes
checking for DATE_TIME_FORMAT_ISO8601... no
checking for THREADS... yes
checking for pthread_setattr_default_np... no
checking for TYPE_INIT... no
checking for WIN32_GET_COMMAND_LINE... yes
checking for STR_TO_ASCII... yes
checking for HAVE_CHECKED_MUL... yes
checking for gtk-doc... no
configure: WARNING:
  You will not be able to create source packages with 'make dist'
  because gtk-doc >= 1.14 is not found.
checking for gtkdoc-check... no
checking for gtkdoc-check... no
checking for gtkdoc-rebase... no
checking for gtkdoc-mkpdf... no
checking whether to build gtk-doc documentation... no
checking for GTKDOC_DEPS... yes
checking for EXPAT... yes
checking for GSF... no
configure: WARNING: libgsf-1 not found; disabling dzsave support
checking for GSF_ZIP64... no
checking for FFTW... no
configure: WARNING: fftw not found; disabling fftw support
checking for MAGICK_WAND... no
checking for IMAGE_MAGICK... no
configure: WARNING: neither MagickCore nor ImageMagick found; disabling Magick support
checking for ImportImagePixels... no
checking for ImagesToBlob... no
checking for ORC... no
configure: WARNING: orc-0.4.11 or later not found; disabling orc support
checking for LCMS... no
configure: WARNING: lcms2 not found; disabling ICC profile support
checking for OPENEXR... no
configure: WARNING: OpenEXR not found; disabling OpenEXR support
checking for X... libraries , headers 
checking for gethostbyname... yes
checking for connect... yes
checking for remove... yes
checking for shmat... yes
checking for IceConnectionNumber in -lICE... yes
checking for NIFTI... libraries (none), headers (none)
checking for HEIF... no
configure: WARNING: libheif >= 1.3.0 not found; disabling HEIF support
checking for PDFIUM... libraries (none), headers (none)
checking for POPPLER... no
configure: WARNING: poppler-glib >= 0.16.0 or cairo >= 1.2 not found; disabling PDF load via poppler
checking for RSVG... yes
checking for ZLIB... yes
checking for OPENSLIDE... no
configure: OpenSlide >= 3.4.0 not found; checking for >= 3.3.0
checking for OPENSLIDE... no
configure: WARNING: OpenSlide >= 3.3.0 not found; disabling virtual slide support
checking for MATIO... no
configure: WARNING: matio not found; disabling matio support
checking for CFITSIO... no
configure: WARNING: cfitsio not found; disabling cfitsio support
checking for LIBWEBP... no
configure: WARNING: libwebp, mux, demux not found; disabling WEBP support
checking for PANGOFT2... yes
checking for TIFF... yes
checking whether COMPRESSION_WEBP is declared... no
checking for giflib... libraries (none), headers (none)
configure: WARNING: giflib not found; disabling direct GIF support
checking for SPNG... no
checking for SPNG... no
checking for PNG... yes
checking for png_set_chunk_malloc_max... yes
checking for IMAGEQUANT... no
configure: WARNING: libimagequant not found; disabling 8bpp PNG support
checking for JPEG... yes
checking for jpeg_c_bool_param_supported... no
checking for EXIF... no
configure: WARNING: libexif >= 0.6 not found; disabling exif support
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating vips.pc
config.status: creating vips-cpp.pc
config.status: creating Makefile
config.status: creating ./libvips/include/vips/version.h
config.status: creating libvips/include/Makefile
config.status: creating libvips/include/vips/Makefile
config.status: creating libvips/Makefile
config.status: creating libvips/arithmetic/Makefile
config.status: creating libvips/colour/Makefile
config.status: creating libvips/colour/profiles/Makefile
config.status: creating libvips/conversion/Makefile
config.status: creating libvips/convolution/Makefile
config.status: creating libvips/deprecated/Makefile
config.status: creating libvips/foreign/Makefile
config.status: creating libvips/freqfilt/Makefile
config.status: creating libvips/histogram/Makefile
config.status: creating libvips/draw/Makefile
config.status: creating libvips/iofuncs/Makefile
config.status: creating libvips/morphology/Makefile
config.status: creating libvips/mosaicing/Makefile
config.status: creating libvips/create/Makefile
config.status: creating libvips/resample/Makefile
config.status: creating cplusplus/include/Makefile
config.status: creating cplusplus/include/vips/Makefile
config.status: creating cplusplus/Makefile
config.status: creating tools/Makefile
config.status: creating tools/batch_crop
config.status: creating tools/batch_image_convert
config.status: creating tools/batch_rubber_sheet
config.status: creating tools/light_correct
config.status: creating tools/shrink_width
config.status: creating test/Makefile
config.status: creating test/variables.sh
config.status: creating test/test-suite/Makefile
config.status: creating test/test-suite/helpers/Makefile
config.status: creating man/Makefile
config.status: creating doc/Makefile
config.status: creating doc/libvips-docs.xml
config.status: creating po/Makefile.in
config.status: creating fuzz/Makefile
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands
config.status: executing default-1 commands
* build options
native win32:				no
native OS X:				no
open files in binary mode: 		no
enable debug:				no
enable deprecated library components:	yes
enable docs with gtkdoc:		no
gobject introspection: 			yes
enable radiance support: 		yes
enable analyze support: 		yes
enable PPM support: 			yes

* optional dependencies
use fftw3 for FFT: 			no
Magick package: 			none
Magick API version: 			none
load with libMagick: 			no
save with libMagick: 			no
accelerate loops with orc: 		no
  (requires orc-0.4.11 or later)
ICC profile support with lcms: 		no
file import with niftiio: 		no
file import with libheif: 		no
file import with OpenEXR: 		no
file import with OpenSlide:		no
  (requires openslide-3.3.0 or later)
file import with matio: 		no
PDF import with PDFium: 		no
PDF import with poppler-glib: 		no
  (requires poppler-glib 0.16.0 or later)
SVG import with librsvg-2.0: 		yes
  (requires librsvg-2.0 2.34.0 or later)
zlib: 					yes
file import with cfitsio: 		no
file import/export with libwebp:	no
  (requires libwebp, libwebpmux, libwebpdemux 0.6.0 or later)
text rendering with pangoft2: 		yes
file import/export with libspng: 	no
  (requires libspng-0.6 or later)
file import/export with libpng: 	yes (pkg-config libpng >= 1.2.9)
  (requires libpng-1.2.9 or later)
support 8bpp PNG quantisation:		no
  (requires libimagequant)
file import/export with libtiff:	yes (pkg-config libtiff-4)
file import/export with giflib:		no
file import/export with libjpeg:	yes (pkg-config)
image pyramid export:			no
  (requires libgsf-1 1.14.26 or later)
use libexif to load/save JPEG metadata: no

**Reply by jcupitt** [reply]
That all looks OK, though you are missing quite a few useful libraries. Just run "make" and it should work.

**Reply by LITTLEKKKK** [reply]
After I solved the system clock problem, still exist "leaving directory" error.
I search for this problem, but can't solve.

make[4]: Entering directory `/public/software/libvips/vips-8.10.5/test/test-suite'
make[4]: Nothing to be done for `all-am'.
make[4]: Leaving directory `/public/software/libvips/vips-8.10.5/test/test-suite'
make[3]: Leaving directory `/public/software/libvips/vips-8.10.5/test/test-suite'
make[3]: Entering directory `/public/software/libvips/vips-8.10.5/test'
make[3]: Nothing to be done for `all-am'.
make[3]: Leaving directory `/public/software/libvips/vips-8.10.5/test'
make[2]: Leaving directory `/public/software/libvips/vips-8.10.5/test'
Making all in fuzz
make[2]: Entering directory `/public/software/libvips/vips-8.10.5/fuzz'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/public/software/libvips/vips-8.10.5/fuzz'
make[2]: Entering directory `/public/software/libvips/vips-8.10.5'
make[2]: Leaving directory `/public/software/libvips/vips-8.10.5'
make[1]: Leaving directory `/public/software/libvips/vips-8.10.5'

**Reply by jcupitt** [reply]
I think that's all OK, "leaving directory" is not an error.

**Reply by LITTLEKKKK** [reply]
vips: unknown action "dzsave"


raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['vips', 'dzsave', 'xxx', 'xxx' .etc]' returned non-zero exit status 1.


It seems be solved. But when I used it, the error above occurred.

**Reply by jcupitt** [reply]
You need libgsf for dzsave support. It's probably called something like `libgsf-1-devel`. You'll need to reconfigure and rebuild libvips, of course.

Which version of centos is this? You might just be able to install libvips with yum, rather than rebuilding from source.

**Reply by LITTLEKKKK** [reply]
> You need libgsf for dzsave support. It's probably called something like `libgsf-1-devel`. You'll need to reconfigure and rebuild libvips, of course.
> 
> Which version of centos is this? You might just be able to install libvips with yum, rather than rebuilding from source.

CentOS7.6
when I use "yum install libvips", it says
No package libvips available.
Error: Nothing to do

so I choose to install from source.

**Reply by jcupitt** [reply]
Ah, gotcha. 

I checked and it's called `libgsf-devel`.

**Reply by jcupitt** [reply]
I made you a dockerfile that installs current libvips on centos7:

https://github.com/jcupitt/docker-builds/blob/master/libvips-centos7/Dockerfile

**Reply by kmartinez** [reply]
last year people started reporting these messages - and I think it was a small change in behaviour in make itself ;-) 
As john says - its harmless info
